### PR TITLE
Feat/swarinfo 362 teleop simulation

### PR DESCRIPTION
**Pull request by g-lesssard**
Simple changes for network + propolis bumped

### CODE REVIEW

File: src/message-handler/src/MessageDispatcher.cpp [hunk 1]
```diff
@@ -61,12 +63,19 @@ bool MessageDispatcher::dispatchNetworkAPI(const MessageDTO& message,
...
         return m_hivemindOutputQueue.push(message);
     }
 
     if (message.getDestinationId() == 0) {
+        m_logger.log(LogLevel::Info, "Forwarded message to broadcast");
         return m_broadcastOutputQueue.push(message);
     }
     // Maybe add check that ip table contains the robot ID in the future.
-    return m_unicastOutputQueue.push(message);
+    if (m_manager.getIPFromAgentID(message.getDestinationId()).has_value()) {
```
Comment (xgroleau) [inline src/message-handler/src/MessageDispatcher.cpp:75]: Is the comment still relevent?

File: src/network/src/common/src/AbstractNetworkManager.cpp [hunk 2]
```diff
@@ -9,12 +9,11 @@ std::optional<uint32_t> AbstractNetworkManager::getIPFromAgentID(uint16_t agentI
 }
 
 bool AbstractNetworkManager::registerAgent(uint16_t agentID, uint32_t ip) {
-    if (m_hashMap.at(agentID).has_value() &&
-        m_hashMap.upsert(std::pair<uint16_t, uint16_t>(agentID, ip))) {
-        m_logger.log(LogLevel::Info, "Updated port of agent %d with value %d", agentID, ip);
+    if (m_hashMap.at(agentID).has_value() && m_hashMap.upsert(agentID, ip)) {
+        // m_logger.log(LogLevel::Info, "Updated port of agent %d with value %d", agentID, ip);
```
Comment (xgroleau) [inline src/network/src/common/src/AbstractNetworkManager.cpp:13]: Still needed?

File: src/main.cpp [hunk 3]
```diff
@@ -202,6 +206,7 @@ class BroadcastMessageDispatcher : public AbstractTask<3 * configMINIMAL_STACK_S
             if (!dispatcher.deserializeAndDispatch()) {
                 m_logger.log(LogLevel::Error, "Fail to deserialize/dispatch broadcast");
             }
+            Task::delay(50);
```
Comment (xgroleau) [inline src/main.cpp:209]: Alot of delay were added, jsute to be sure, are they supposed to still be here or it was for debugging?

File: src/main.cpp [hunk 4]
```diff
@@ -108,6 +109,7 @@ class UnicastMessageSenderTask : public AbstractTask<3 * configMINIMAL_STACK_SIZ
             if (!messageSender.processAndSerialize()) {
```
Comment (xgroleau) [inline src/main.cpp:109]: Should we check if the socket is still connected to know if it's a connection that changed vs a fail to deserialize?

**Review by xgroleau — APPROVED** [review]

File: src/main.cpp (hunk 3 above)
Comment (g-lesssard) [reply src/main.cpp:209]: I added them for safety, to make sure the queues are not locked by reading them with a mutex

**Review by xgroleau — APPROVED** [review]

**Review by xgroleau — CHANGES_REQUESTED** [review]
Couple minor comments

File: src/network/src/common/src/AbstractNetworkManager.cpp [hunk 5]
```diff
@@ -9,12 +9,10 @@ std::optional<uint32_t> AbstractNetworkManager::getIPFromAgentID(uint16_t agentI
...
 
 bool AbstractNetworkManager::registerAgent(uint16_t agentID, uint32_t ip) {
-    if (m_hashMap.at(agentID).has_value() &&
-        m_hashMap.upsert(std::pair<uint16_t, uint16_t>(agentID, ip))) {
-        m_logger.log(LogLevel::Info, "Updated port of agent %d with value %d", agentID, ip);
+    if (m_hashMap.at(agentID).has_value() && m_hashMap.upsert(agentID, ip)) {
         return true;
     }
-    if (m_hashMap.insert(std::pair<uint16_t, uint16_t>(agentID, ip))) {
+    if (m_hashMap.insert(agentID, ip)) {
```
[the remark this answers is not in the document]
Comment (g-lesssard) [reply src/network/src/common/src/AbstractNetworkManager.cpp:15]: I made it this way to log the update vs a new agent

**Review by xgroleau — APPROVED** [review]

**Pull request merged**